com.unity.multiplayer.mlapi/Runtime/Core/NetworkedBehaviour.cs

```diff
@@ -67,24 +67,40 @@ public BitSerializer __beginSendServerRpc(ServerRpcParams serverRpcParams, bool
 #endif
         {
             var rpcQueueContainer = NetworkingManager.Singleton.rpcQueueContainer;
-            var writer = rpcQueueContainer.BeginAddQueueItemToOutboundFrame(
-                RpcQueueContainer.QueueItemType.ServerRpc,
-                Time.realtimeSinceStartup,
-                Transport.MLAPI_STDRPC_CHANNEL,
-                /* sendFlags = */ 0,
-                NetworkingManager.Singleton.ServerClientId,
-                /* targetNetworkIds = */ null);
-            if (!rpcQueueContainer.IsUsingBatching())
+            PooledBitWriter writer;
+            var IsUsingBatching = rpcQueueContainer.IsUsingBatching();
+            if (IsHost)
             {
+                if (serverRpcParams.Send.UpdateStage == NetworkUpdateManager.NetworkUpdateStage.Default)
+                {
+                    serverRpcParams.Send.UpdateStage = NetworkUpdateManager.NetworkUpdateStage.Update;
+                }
+                writer = rpcQueueContainer.BeginAddQueueItemToFrame(RpcQueueContainer.QueueItemType.ServerRpc, Time.realtimeSinceStartup, Transport.MLAPI_STDRPC_CHANNEL, 0,
+                    NetworkingManager.Singleton.ServerClientId, null, QueueHistoryFrame.QueueFrameType.Inbound,serverRpcParams.Send.UpdateStage );
+
+                //Under this condition we always treat this like there is no batching
```

Comment:
Is this because technically we don't need to batch when in host (vs. server) mode?  Or because batching doesn't work in host mode?  My thinking is, I really would like to avoid the host mode special case code in general because I think we tend to test one or the other leading to latent bugs.  If we don't need to special case host mode then could we meld the if and else clauses into one?